pkg/objstore/s3/s3.go

```diff
@@ -31,13 +32,25 @@ const DirDelim = "/"
 
 // Config stores the configuration for s3 bucket.
 type Config struct {
-	Bucket        string `yaml:"bucket"`
-	Endpoint      string `yaml:"endpoint"`
-	AccessKey     string `yaml:"access_key"`
-	Insecure      bool   `yaml:"insecure"`
-	SignatureV2   bool   `yaml:"signature_version2"`
-	SSEEncryption bool   `yaml:"encrypt_sse"`
-	SecretKey     string `yaml:"secret_key"`
+	Bucket        string     `yaml:"bucket"`
+	Endpoint      string     `yaml:"endpoint"`
+	AccessKey     string     `yaml:"access_key"`
+	Insecure      bool       `yaml:"insecure"`
+	SignatureV2   bool       `yaml:"signature_version2"`
+	SSEEncryption bool       `yaml:"encrypt_sse"`
+	SecretKey     string     `yaml:"secret_key"`
+	HTTPConfig    HTTPConfig `yaml:"http_config"`
+}
+
+// HTTPConfig stores the http.Transport configuration for the s3 minio client
+type HTTPConfig struct {
+	IdleConnTimeout       model.Duration `yaml:"idle_conn_timeout"`
```

Comment:
I would really suggest just doing the thing you need right now which is IdleConnTimeout explicitly for HA Proxy.

I feel we should stick to YAGNI here and not introduce something that is not useful right now: https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it